### ISSUE TITLE
[unbound-method] false negative when destructuring a method

### ISSUE THREAD

**Issue by hgezim** [typescript-eslint/typescript-eslint#2198]
**Repro**

```JSON
{
  "rules": {
    
    "@typescript-eslint/no-explicit-any": "off",
    "@typescript-eslint/no-unused-vars": "off",
    "@typescript-eslint/no-inferrable-types": "off",
    "@typescript-eslint/naming-convention": [
      "error",
      { "selector": "variableLike", "format": ["camelCase", "UPPER_CASE", "PascalCase"], "leadingUnderscore": "allow" },
    ],
    "@typescript-eslint/unbound-method": ["error"],
  }
}
```

```TS
class MyClass {
    public log(): void {
        console.log(this);
    }
}

const instance = new MyClass();

// This logs the global scope (`window`/`global`), not the class instance
const myLog = instance.log;
myLog();

// This log might later be called with an incorrect scope
const { log } = instance;

log()
```

**Expected Result**
`const { log } = instance;` to show error.

**Actual Result**
`const { log } = instance;` doesn't show any errors.

`const myLog = instance.log;` does error as expected.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.2.0` |
| `@typescript-eslint/parser`        | `3.2.0` |
| `TypeScript`                       | `3.9.5` |
| `ESLint`                           | `7.2.0` |
| `node`                             | `11.15.0` |
| `npm`                              | `6.7.0` |

**Comment by bradzacher** [comment]
it looks like the test for this case is incorrect:
https://github.com/typescript-eslint/typescript-eslint/blob/ee9f100a2f9a874c2b361482742686eeaa9bdac7/packages/eslint-plugin/tests/rules/unbound-method.test.ts#L291-L292

**Comment by bradzacher** [comment]
duplicate of #1112
